examples/maps_heatmaps/src/index.ts

```diff
@@ -126,471 +120,204 @@ function getPoints() {
+  const allpoints : Array<google.maps.LatLng> = [];
+  const images = database.collection("imagesTal");
+  images.get().then(
+     queryImages => { queryImages.forEach(
+        doc=> {prinInfoOnObject(doc);
+          allpoints.push( nowlatlon( doc.get('coordinates').latitude, doc.get('coordinates').longitude ))})});
+    return allpoints  
+}
+
+//a function than returns fillter collection of  all the coordinates in a certain year
+function qeuryByYear(year:number)
+{
+  const allpoints : Array<google.maps.LatLng> = []; 
+  const dataRef:firebase.firestore.Query = database.collection('imagesTal');
+  return dataRef.where('year', '==', year);
+} 
+
+//a function returns a fillter collection that contain coordinates in a certain time from a certain label
+function queryByDateAndLabel(label:string,year?: number, month?: number, day?:number)
+{
+  const allpoints : Array<google.maps.LatLng> = []; 
+  let dataRef:firebase.firestore.Query = database.collection('imagesTal');
+  dataRef = dataRef.where('label', 'array-contains', label);
+  if (year != null)
+    dataRef = dataRef.where('year', '==', year);
+  if (year != null && month!=null)
+    dataRef = dataRef.where('month', '==', month)
+  if (year != null && month!=null && day!=null)
+    dataRef = dataRef.where('day', '==', day);
+  return dataRef;
+}
+
+
+//having truble...
+/*
+function qeuryByGeoPoint()
+{
+  // Create a GeoFirestore reference
+const GeoFirestore = geofirestore.initializeApp(database);
+// Create a GeoCollection reference
+const geocollection = GeoFirestore.collection('imagesTal');
+geocollection.where('year','==',1999).get().then( queryImages => { queryImages.forEach(doc=> console.log(doc.data))})
+
+
+// Create a GeoQuery based on a location
+const query: geofirestore.GeoQuery  = geocollection.near({ center: new firebase.firestore.GeoPoint(37.800811, -122.436275), radius: 10000000000 });
+let allpoints : Array<google.maps.LatLng> = []; 
+// Get query (as Promise)
+query.get().then((value) => {console.log(value)
+
+  // All GeoDocument returned by GeoQuery, like the GeoDocument added above
+  value.docs.forEach(doc =>  {console.log('******3');
+  console.log(doc);
+    console.log(doc.data.arguments.coordinates.latitude());
+    allpoints.push( nowlatlon( doc.data.arguments.coordinates.latitude(), doc.data.arguments.coordinates.longitude() ))})
+  });
+}
+*/
+
+
+//this function recives a map and a filterd colection and add all the geo points to the heatmap
+function getPointsByQeury(heatmap:google.maps.visualization.HeatmapLayer, dataRef :firebase.firestore.Query)
+{
+  const allpoints : Array<google.maps.LatLng> = []; 
+  dataRef.get().then(
+    queryImages => { queryImages.forEach(
+       doc=> {prinInfoOnObject(doc);
+         allpoints.push( nowlatlon( doc.get('coordinates').latitude, doc.get('coordinates').longitude ))})
+         heatmap.setData(allpoints);
+        });
+}
+
+
+
+//creats new geo point
+function nowlatlon(lat:number,lon:number):google.maps.LatLng
```

Comment:
A good helper function :).
How about extending it to `toLatLong(coordinates: firebase.firestore.GeoPoint)`
Also, follow the [style guide for naming](go/tsstyle-oss-interns?cl=320047486#identifiers).